app/assets/javascripts/discourse/app/app.js

```diff
@@ -75,7 +75,42 @@ const Discourse = Application.extend({
       } else if (/\/(api\-)?initializers\//.test(key)) {
         const initializer = this._prepareInitializer(key);
         if (initializer) {
-          this.instanceInitializer(initializer);
+          // FIXME!
```

Comment:
Discourse initializers (including themes/plugins) are guaranteed to be namespaced under `discourse/`, so we could skip any modules outside that.

(core will be `discourse/initializers`, plugins `discourse/plugins/plugin-name/**/initializers`, themes `discourse/theme-{id}/**/initializers`). Totally agree about the naming issues - we're tracking that in [this Meta topic](https://meta.discourse.org/t/updating-our-initializer-naming-patterns/241919?u=david)